### ISSUE TITLE
Bind TCP Ruby shell is broken

### ISSUE THREAD

**Issue by capnspacehook** [0x00-0x00/ShellPop#26]
As I was going though all the bind shells testing for an upcoming PR, I noticed that I could not get the linux/bind/tcp/ruby shell to work. Every time I ran it I got this error from sh:

```
ruby -rsocket -e 'f=TCPServer.new(1337);s=f.accept;exec sprintf("/bin/bash -i <&%d >&%d 2>&%d",s,s,s)'
sh: 1: 8: Bad file descriptor
```

From the looks of it, I'm guessing bash doesn't like `f.accept` as a file descriptor, but not sure how to fix that. My ruby knowledge is very little.

**Comment by 0x00-0x00** [comment]
Hey @capnspacehook, thanks for raising this issue.
Though, I am not able to reproduce this problem, as my ruby bind shell works fine.

This file descriptor problem rises when you try to bind descriptors to /bin/sh instead of /bin/bash. 

Note that the problem resides in file descriptor binding to this section of code:
```
/bin/bash -i <&%d >&%d 2>&%d
```
This works with bash, but not with sh shell. Yet, your interpreter is trying to use /bin/sh for these pipe redirections, so it raises exceptions. 

Are you sure that your /bin/bash is not symlinked or something?

**Comment by 0x00-0x00** [comment]
Commit fix is in 777b17c

**Comment by capnspacehook** [comment]
No problem! Thanks for looking into it so quickly!
